File: Algorithms/9-Palindrome-Number/Palindrome-Number.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        int temp = x, digit = 0, num = 0;
        while (x > 0) {
            digit = x % 10;
            if (num > Integer.MAX_VALUE / 10 || (num == Integer.MAX_VALUE / 10 && digit > 7)) return false;
            if (num < Integer.MIN_VALUE / 10 || (num == Integer.MIN_VALUE / 10 && digit < -8)) return false;
            num = num * 10 +  digit;
            x /= 10;
    }
        return temp == num ? true : false;
    }
}

File: Algorithms/9-Palindrome-Number/Palindrome-Number.c
bool isPalindrome(int x) {
    int temp = x;
    long rem = 0, new = 0;
    if(x < 0){
        return false;
    }
    for(int i = 0; x > 0; i++){
        rem = x % 10;
        new = new * 10 + rem;
        x = x / 10;
    }
    if(temp == new){
        return true;
    }
    return false;
}